scanner/internal/httputil/transport_mux.go

```diff
@@ -0,0 +1,126 @@
+// Package httputil defines utility HTTP functions built specifically for Scanner's use cases.
+//
+// It is possible other StackRox components may benefit from these functions,
+// but that will be considered at a future time.
+package httputil
+
+import (
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	"github.com/stackrox/rox/pkg/clientconn"
+	"github.com/stackrox/rox/pkg/env"
+	"github.com/stackrox/rox/pkg/httputil"
+	"github.com/stackrox/rox/pkg/httputil/proxy"
+	"github.com/stackrox/rox/pkg/mtls"
+)
+
+var (
+	// defaultDialer is essentially copied from http.DefaultTransport from go1.20.10.
+	defaultDialer = net.Dialer{
+		Timeout:   30 * time.Second,
+		KeepAlive: 30 * time.Second,
+	}
+)
+
+type options struct {
+	denySensor bool
```

Comment:
This necessary if its also controlled via `mtls` authorizers and k8s `NetworkPolicy`?

Reply:
If nobody does anything strange, then no, but I figure it doesn't hurt to add an extra layer of safety + logging. In a way it can help warn us during development